cassandra-executor/src/test/java/io/mesosphere/mesos/frameworks/cassandra/CassandraExecutorTest.java

```diff
@@ -0,0 +1,281 @@
+                .setTaskType(CassandraFrameworkProtos.TaskDetails.TaskType.NODE_JOB_STATUS)
+                .build());
+        List<CassandraFrameworkProtos.SlaveStatusDetails> messages = driver.frameworkMessages();
+        assertEquals(1, messages.size());
+        assertTrue(messages.get(0).hasNodeJobStatus());
+        List<Protos.TaskStatus> taskStatusList = driver.taskStatusList();
+        assertEquals(1, taskStatusList.size());
+        assertEquals(taskId, taskStatusList.get(0).getTaskId());
+        assertEquals(Protos.TaskState.TASK_FINISHED, taskStatusList.get(0).getState());
+
+        assertNull(executor.getCurrentJob());
+        assertTrue(objectFactory.storageServiceProxy.listeners.isEmpty());
+    }
+
+    private void repairJob() {
+        CassandraFrameworkProtos.ClusterJobType jobType = CassandraFrameworkProtos.ClusterJobType.REPAIR;
+
+        assertNull(executor.getCurrentJob());
+        assertTrue(objectFactory.storageServiceProxy.listeners.isEmpty());
+
+        Protos.TaskID taskId = Protos.TaskID.newBuilder().setValue(driver.executorInfo.getExecutorId().getValue() + '.' + jobType).build();
+        driver.launchTask(
+                taskId,
+                Protos.CommandInfo.getDefaultInstance(),
+                CassandraFrameworkProtos.TaskDetails.newBuilder()
+                        .setTaskType(CassandraFrameworkProtos.TaskDetails.TaskType.NODE_JOB)
+                        .setNodeJobTask(CassandraFrameworkProtos.NodeJobTask.newBuilder()
+                                .setJobType(jobType))
+                        .build(),
+                "node job task",
+                Collections.<Protos.Resource>emptyList());
+
+        taskStartingRunning(taskId);
+
+        driver.frameworkMessage(CassandraFrameworkProtos.TaskDetails.newBuilder()
+                .setTaskType(CassandraFrameworkProtos.TaskDetails.TaskType.NODE_JOB_STATUS)
+                .build());
+        List<CassandraFrameworkProtos.SlaveStatusDetails> messages = driver.frameworkMessages();
+        assertEquals(1, messages.size());
+        assertTrue(messages.get(0).hasNodeJobStatus());
+        assertTrue(driver.taskStatusList().isEmpty());
+
+        assertNotNull(executor.getCurrentJob());
+        // MBean emits system KS + 3 other KS; one KS is currently processing; 2 KS remaining
+        assertEquals(2, executor.getCurrentJob().getRemainingKeyspaces().size());
+        assertFalse(executor.getCurrentJob().isFinished());
+        assertFalse(objectFactory.storageServiceProxy.listeners.isEmpty());
+
+        objectFactory.storageServiceProxy.emitRepairNotification(ActiveRepairService.Status.STARTED);
+        objectFactory.storageServiceProxy.emitRepairNotification(ActiveRepairService.Status.SESSION_SUCCESS);
+        objectFactory.storageServiceProxy.emitRepairNotification(ActiveRepairService.Status.SESSION_SUCCESS);
+        objectFactory.storageServiceProxy.emitRepairNotification(ActiveRepairService.Status.SESSION_SUCCESS);
```

Comment:
What is the significance to calling this three times with the same argument? Please add a comment explaining.